src/modules/menu/components/advancedTools/ImportFile/utils.js

```diff
@@ -39,31 +41,59 @@ export function isGpx(fileContent) {
  * @param {OBject} store Vuex store
  * @param {ArrayBuffer} content Content of the file
  * @param {string} source Source of the file (either URL or file path)
+ * @param {File | null} [originalFile]
  * @returns {ExternalLayer} External layer object
  */
-export async function handleFileContent(store, content, source) {
+export async function handleFileContent(store, content, source, originalFile = null) {
     let layer = null
-    let textContent
+    let parsedContent
     let linkFiles
     if (isZipContent(content)) {
         log.debug(`File content is a zipfile, assume it is a KMZ archive`)
         const kmz = await unzipKmz(content, source)
-        textContent = kmz.kml
+        parsedContent = kmz.kml
         linkFiles = kmz.files
+    } else if (isTiffContent(content)) {
+        log.debug(`File content might be a GeoTIFF, attempting a parse as such`)
+        try {
+            parsedContent = await fromArrayBuffer(content)
+            layer = new GeoTIFFLayer({
+                fileSource: source,
+                visible: true,
+                opacity: 1.0,
+                // For local files : OpenLayers needs a Blob, and not an already parsed GeoTIFF instance
+                data: originalFile,
+            })
+            store.dispatch('addLayer', { layer, ...dispatcher })
+            // looking into the GeoTIFF file to get the extent
+            const geoTIFFImage = await parsedContent.getImage()
+            const geoTIFFExtent = geoTIFFImage.getBoundingBox()
+            if (geoTIFFExtent) {
+                store.dispatch('zoomToExtent', {
+                    extent: normalizeExtent(geoTIFFExtent),
+                    ...dispatcher,
+                })
+            }
```

Comment:
What if extent is not within switzerland, should we not do the same as for KML and GPX here ?

Reply:
good point, yes we should do the same logic.
I'll try to gather all extent use case together